src/access/AccessControl.cpp

```diff
@@ -33,8 +33,17 @@ using namespace chip::Access;
 
 namespace {
 
-AccessControl defaultAccessControl;
-AccessControl * globalAccessControl = &defaultAccessControl;
+AccessControl * sGetDefaultAccessControl()
+{
+    static auto* defaultAccessControl = new AccessControl{};
```

Comment:
Embedded platforms generally like static allocation so that compile fails rather than potentially memory fragmentation and running out of memory at runtime.

I think we should figure out and agree on a path forward here, with feedback from folks that worked on a lot of these small devices.

Reply:
I've worked on a few of these small devices and points listed here are valid. Dynamic memory allocation is definitely frowned upon as a general rule of thumb. There are of course exceptions to that case, but for the most part its avoided as much as possible so that the system can remain deterministic.